pkg/action/action.go

```diff
@@ -197,8 +214,65 @@ func (c *Configuration) recordRelease(r *release.Release) {
 	}
 }
 
-type RESTClientGetter interface {
-	ToRESTConfig() (*rest.Config, error)
-	ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error)
-	ToRESTMapper() (meta.RESTMapper, error)
+// InitActionConfig initializes the action configuration
+func InitActionConfig(envSettings *cli.EnvSettings, allNamespaces bool, helmDriver string, log DebugLog) (*Configuration, error) {
+
+	var actionConfig Configuration
+	kubeconfig := kubeConfig(envSettings)
+
+	kc := kube.New(kubeconfig)
+	kc.Log = log
+
+	clientset, err := kc.Factory.KubernetesClientSet()
+	if err != nil {
+		return nil, err
+	}
+	var namespace string
+	if !allNamespaces {
+		namespace = GetNamespace(envSettings)
+	}
+
+	var store *storage.Storage
+	switch helmDriver {
+	case "secret", "secrets", "":
+		d := driver.NewSecrets(clientset.CoreV1().Secrets(namespace))
+		d.Log = log
+		store = storage.Init(d)
+	case "configmap", "configmaps":
+		d := driver.NewConfigMaps(clientset.CoreV1().ConfigMaps(namespace))
+		d.Log = log
+		store = storage.Init(d)
+	case "memory":
+		d := driver.NewMemory()
+		store = storage.Init(d)
+	default:
+		// Not sure what to do here.
+		panic("Unknown driver in HELM_DRIVER: " + helmDriver)
+	}
+
+	actionConfig.RESTClientGetter = kubeconfig
+	actionConfig.KubeClient = kc
+	actionConfig.Releases = store
+	actionConfig.Log = log
+
+	return &actionConfig, nil
+}
+
+func kubeConfig(envSettings *cli.EnvSettings) genericclioptions.RESTClientGetter {
+	configOnce.Do(func() {
+		config = kube.GetConfig(envSettings.KubeConfig, envSettings.KubeContext, envSettings.Namespace)
+	})
+	return config
+}
+
+//GetNamespace gets the namespace from the configuration
+func GetNamespace(envSettings *cli.EnvSettings) string {
```

Comment:
Can you make this part of the `*cli.EnvSettings` object? e.g. in `pkg/cli`:

```
func (e *EnvSettings) Namespace() string {
    if e.Namespace != "" {
        return e.Namespace
    }

    if ns, _, err := e.kubeConfig().ToRawKubeConfigLoader().Namespace(); err != nil {
        return ns
    }

    return "default"
}
```

Reply:
I think I captured the intent here.